package/MDAnalysis/topology/PDBParser.py

```diff
@@ -288,10 +290,6 @@ def _parseatoms(self):
             warnings.warn("Serial numbers went over 100,000.  "
                           "Higher serials have been guessed")
 
-        # If segids not present, try to use chainids
-        if not any(segids):
-            segids = chainids
```

Comment:
If you put this back in, that second and third failing tests should pass. I think that this is expected behaviour from most users, that differences in chain are conceptually similar to differences in segment -- better not to change this part.